### PR TITLE
Interop plans

### PR DESCRIPTION
**Pull request by not4Pedro**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: .github/workflows/interop-tests.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        run: |
+          {
+              echo 'FROM fedora:latest'
+              echo 'RUN dnf -y update'
+              echo 'RUN dnf -y install perl'
+              echo 'RUN dnf -y install git tmt-all'
+              echo 'RUN dnf -y remove openssl'
+              echo 'RUN mkdir /root/bin'
+              echo "RUN echo '#!/bin/bash /openssl/util/wrap.pl /openssl/apps/openssl' > /root/bin/openssl & chmod +x /root/bin/openssl"
+              echo 'RUN git clone -b interop-plans --recurse-submodules https://github.com/not4pedro/openssl.git'
```
Comment (paulidale) [inline .github/workflows/interop-tests.yml:36]: Does this need --recurse-sumbodules ?
That results in fairly expensive tests that aren't going to be much of a benefit IMO.

File: .github/workflows/interop-tests.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+              echo 'WORKDIR /openssl/'
+              echo 'RUN ./config --banner=Configured no-asm no-makedepend enable-buildtest-c++ enable-fips --strict-warnings -D_DEFAULT_SOURCE && perl configdata.pm --dump'
```
Comment (paulidale) [inline .github/workflows/interop-tests.yml:38]: What is the reason for `-D_DEFAULT_SOURCE` here?

I'd drop `enable-buildtest-c++` and possibly `no-asm` (less sure about this, it depends on what you are trying to test).

File: plans/gnutls-2way.fmf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+summary: Upstreamed interop-2way tests
+contact: Stanislav Zidek <szidek@redhat.com>
+discover:
+  # upstreamed tests (public)
+  - name: interop-gnutls-2way
+    how: fmf
+    url: https://gitlab.com/redhat-crypto/tests/interop.git
+    filter: 'tag: interop-openssl & tag: interop-gnutls & tag: interop-2way & tag: -interop-nss-slow'
+execute:
+    how: tmt
```
Comment (paulidale) [inline plans/gnutls-2way.fmf:10]: Missing newline at end of file -- here and elsewhere.

File: .github/workflows/interop-tests.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+name: Interoperability tests with GnuTLS and NSS
+on: [pull_request, push]
```
Comment (paulidale) [inline .github/workflows/interop-tests.yml:2]: Depending on test duration, this might switch to daily or push.

**Review by paulidale — COMMENTED** [review]
This looks reasonable.

**Comment by ep69** [comment]
I managed to get the NSS interop tests running by adding `--sysctl net.ipv6.conf.lo.disable_ipv6=0` to `docker run` (see [my experimental PR](https://github.com/ep69/openssl/pull/1/files#diff-50cffd505fe798970f550c9e84e9edc6036275a29193dd39cb2a1b7c18f90757R49)).

Other things that I think could be improved:

1. make shallow  copies with `git clone` and `git submodule update` (add `--depth=1`)
2. look into possibilities of pre-building the docker image in some registry to save tim

**Comment by not4Pedro** [comment]
Hi @ep69,
Thanks!
I have checked the possibilities and this time I went for the GitHub Container Registry, it could be an option to consider. Currently, the image resides [here](https://github.com/not4Pedro?tab=packages)

We should make the image source (docker file) public too. Will we just upload the source in the repo for reference? 
Docker file:
```
FROM fedora:latest
RUN dnf -y update
RUN dnf -y install perl
RUN dnf -y install git tmt-all
RUN dnf -y remove openssl
RUN mkdir /root/bin
RUN echo '#!/bin/bash /openssl/util/wrap.pl /openssl/apps/openssl' > /root/bin/openssl & chmod +x /root/bin/openssl
RUN git clone --depth=1 -b interop-plans https://github.com/not4pedro/openssl.git
WORKDIR /openssl/
RUN  git submodule update --depth=1 --init interop
RUN ./config --banner=Configured no-asm no-makedepend enable-fips --strict-warnings && perl configdata.pm --dump
RUN make -s -j4
WORKDIR /openssl/interop
```
In terms of layers of this docker file, I see there is still room for improvement too.

**Comment by ep69** [comment]
I think the best next step would be to create an official PR in openssl project. I am pretty sure they won't like a dependency on the image registry in your personal profile, so make the first version as is with building the image on the fly and suggest to use openssl registry as a speed improvement, if desired.

**Pull request closed without merging**